Examples/Algorithms/Digitization/include/ActsExamples/Digitization/Smearers.hpp

```diff
@@ -0,0 +1,280 @@
+  /// @return a Result is uniformly distributed between bin borders
+  Acts::Result<std::pair<double, double>> operator()(double value,
+                                                     RandomEngine& rnd) {
+    if (binningData.min < value and binningData.max > value) {
+      auto bin = binningData.search(value);
+      auto lower = binningData.boundaries()[bin];
+      auto higher = binningData.boundaries()[bin + 1];
+      double svalue = lower + (higher - lower) * dist(rnd);
+      return Acts::Result<std::pair<double, double>>(
+          std::pair<double, double>(svalue, (higher - lower) / std::sqrt(12.)));
+    }
+    return ActsFatras::DigitizationError::SmearingError;
+  }
+};
+
+/// Digital emulation of a single parameter
+///
+/// It estimates the bin and gives the bin center value
+struct Digital {
+  Acts::BinningData binningData;
+
+  /// Construct with a @param pitch standard deviation and @param range
+  Digital(double pitch, const std::pair<double, double>& range_)
+      : binningData(Acts::open, Acts::binX,
+                    (range_.second - range_.first) / pitch, range_.first,
+                    range_.second) {}
+
+  /// Constructor with a bin utility in order to get the bin borders
+  ///
+  /// @param bu the bin utility within hich the parameter is allowed
+  Digital(Acts::BinningData&& bd) : binningData(std::move(bd)) {}
+
+  /// Call operator for the SmearFunction caller interface
+  ///
+  /// @param value parameter to be smeared
+  /// @param rnd random generator to be used for the call (unused)
+  ///
+  /// @return a Result is uniformly distributed between bin borders
+  Acts::Result<std::pair<double, double>> operator()(double value,
+                                                     RandomEngine& /*unused*/) {
+    if (binningData.min < value and binningData.max > value) {
+      auto bin = binningData.search(value);
+      auto lower = binningData.boundaries()[bin];
+      auto higher = binningData.boundaries()[bin + 1];
+      double svalue = 0.5 * (lower + higher);
+      return Acts::Result<std::pair<double, double>>(
+          std::pair<double, double>(svalue, (higher - lower) / std::sqrt(12.)));
+    }
+    return ActsFatras::DigitizationError::SmearingError;
+  }
+};
+
+template <size_t DIM>
+using Functions = std::array<ActsFatras::SmearFunction<RandomEngine>, DIM>;
+
+template <size_t DIM>
+using Types = std::array<int, DIM>;
+
+template <size_t DIM>
+using Parameters = std::array<std::vector<double>, DIM>;
+
+/// Struct to generate smearing functions from arguments
+///
+struct FunctionGenerator {
+  /// Template unrolling
+  ///
+  /// @tparam DIM The dimension of the smearing function array
+  /// @tparam ENTRY is the entry that is currently filled
+  ///
+  /// @param functions[in,out] The smearing functions that are generated
+  /// @param types The smearing function type (Gauss as default)
+  /// @param pars The parameters for the smearing function
+  template <size_t DIM, size_t ENTRY>
```

Comment:
```suggestion
  template <size_t kDim, size_t kIndex>
```
To conform to our naming guidelines.

Reply:
Fixed.